test/integration/migrate/migration-integration-tests.js

```diff
@@ -665,71 +672,122 @@ module.exports = function (knex) {
-                  .then((data) => {
-                    expect(data).to.have.length(0);
-                  });
-              });
-          });
+        afterEach(() => {
+          return knex.migrate.rollback(
+            { directory: ['test/integration/migrate/test'] },
+            true
+          );
+        });
+
+        it('should only undo the last migration that was run if all migrations have run', function () {
+          return knex.migrate
+            .down({
+              directory: ['test/integration/migrate/test'],
+            })
+            .then(() => {
+              return knex('knex_migrations')
+                .select('*')
+                .then((data) => {
+                  expect(data).to.have.length(1);
+                  expect(path.basename(data[0].name)).to.equal(
+                    '20131019235242_migration_1.js'
+                  );
+                });
+            });
+        });
+
+        it('should only undo the last migration that was run if there are other migrations that have not yet run', function () {
+          return knex.migrate
+            .down({
+              directory: ['test/integration/migrate/test'],
+            })
+            .then(() => {
+              return knex.migrate
+                .down({
+                  directory: ['test/integration/migrate/test'],
+                })
+                .then(() => {
+                  return knex('knex_migrations')
+                    .select('*')
+                    .then((data) => {
+                      expect(data).to.have.length(0);
+                    });
+                });
+            });
+        });
+
+        it('should not error if all migrations have already been undone', function () {
+          return knex.migrate
+            .rollback({ directory: ['test/integration/migrate/test'] }, true)
+            .then(() => {
+              return knex.migrate
+                .down({
+                  directory: ['test/integration/migrate/test'],
+                })
+                .then((data) => {
+                  expect(data).to.be.an('array');
+                });
+            });
+        });
       });
 
-      it('should not error if all migrations have already been undone', function () {
-        return knex.migrate
-          .rollback({ directory: ['test/integration/migrate/test'] }, true)
-          .then(() => {
-            return knex.migrate
-              .down({
-                directory: ['test/integration/migrate/test'],
-              })
-              .then((data) => {
-                expect(data).to.be.an('array');
-              });
+      describe('with transactions disabled', () => {
+        afterEach(async () => {
+          await knex.migrate.rollback(
+            {
+              directory:
+                'test/integration/migrate/test_single_per_migration_trx_disabled',
+            },
+            true
+          );
+        });
+
+        it('should run', async () => {
```

Comment:
does this test those migrations to simply not fail with an error? can we also assert that they actually did what we expect them to?

Reply:
The migrations don't actually do anything other than throw an error if called within a transaction.  Added a comment to that effect.